macros/light/src/traits.rs

```diff
@@ -198,46 +198,55 @@ fn process_fields_and_attributes(name: &Ident, fields: FieldsNamed) -> TokenStre
-                fn get_authority(&self) -> &AccountInfo<'info> {
+                fn get_authority(&self) -> &::anchor_lang::prelude::AccountInfo<'info> {
                     &self.#authority_field
                 }
             }
-            impl<'info> LightSystemAccount<'info> for #name<'info> {
-                fn get_light_system_program(&self) -> &Program<'info, LightSystemProgram> {
+            impl<'info> ::light_sdk::traits::LightSystemAccount<'info> for #name<'info> {
+                fn get_light_system_program(&self) -> &::anchor_lang::prelude::Program<
+                    'info,
+                    ::light_system_program::program::LightSystemProgram
+                > {
                     &self.#light_system_program_field
                 }
             }
         };
         let invoke_accounts_impl = quote! {
-            impl<'info> InvokeAccounts<'info> for #name<'info> {
-                fn get_registered_program_pda(&self) -> &Account<'info, RegisteredProgram> {
+            impl<'info> ::light_sdk::traits::InvokeAccounts<'info> for #name<'info> {
+                fn get_registered_program_pda(&self) -> &::anchor_lang::prelude::Account<
+                    'info,
+                    ::account_compression::RegisteredProgram
+                > {
                     &self.#registered_program_pda_field
                 }
-                fn get_noop_program(&self) -> &AccountInfo<'info> {
+                fn get_noop_program(&self) -> &::anchor_lang::prelude::AccountInfo<'info> {
                     &self.#noop_program_field
                 }
-                fn get_account_compression_authority(&self) -> &AccountInfo<'info> {
+                fn get_account_compression_authority(&self) -> &::anchor_lang::prelude::AccountInfo<'info> {
                     &self.#account_compression_authority_field
                 }
-                fn get_account_compression_program(&self) -> &Program<'info, AccountCompression> {
+                fn get_account_compression_program(&self) -> &::anchor_lang::prelude::Program<
+                    'info,
+                    ::account_compression::program::AccountCompression
+                > {
                     &self.#account_compression_program_field
                 }
-                fn get_system_program(&self) -> &Program<'info, System> {
+                fn get_system_program(&self) -> &::anchor_lang::prelude::Program<'info, System> {
                     &self.#system_program_field
                 }
-                fn get_compressed_sol_pda(&self) -> Option<&UncheckedAccount<'info>> {
+                fn get_compressed_sol_pda(&self) -> Option<&::anchor_lang::prelude::UncheckedAccount<'info>> {
                     #compressed_sol_pda_field
                 }
-                fn get_compression_recipient(&self) -> Option<&UncheckedAccount<'info>> {
+                fn get_compression_recipient(&self) -> Option<&::anchor_lang::prelude::UncheckedAccount<'info>> {
```

Comment:
can we change the traits to return `AccountInfo<'info>` or `Option<AccountInfo<'info>` instead of `&UncheckedAccount` and `Option<`UncheckedAccount`?
To use these in the cpi function we will need to unwrap and do `.to_account_info()` to transfer ownership since cpi invocations need to own all data involved and don't work with references.

Reply:
see traits in my[ registry pr](https://github.com/Lightprotocol/light-protocol/pull/1038/files#diff-1bf83dcafc4b411071970d9450fb1085ac68b5132f761694f16aa6470686f1e4R1-R37) for reference

Reply:
Done